collector_test.go

```diff
@@ -1,279 +1,121 @@
 package sync
 
 import (
+	"context"
+	"iter"
 	"sync"
-	"sync/atomic"
 	"testing"
 	"time"
 
-	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
 	"github.com/anchore/go-sync/internal/stats"
 )
 
-func Test_Collectors(t *testing.T) {
-	for i := 0; i < 100; i++ {
-		Test_Collector(t)
+func Test_CollectCancelRepeat(t *testing.T) {
```

Comment:
might be worth commenting the motivation behind this test -- it isn't clear what were defending against here since there are no shared resources between these test calls (maybe there is and I'm missing something?)